lib/Collaboration/Reference/TalkReferenceProvider.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Joas Schilling <coding@schilljs.com>
+ *
+ * @author Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Talk\Collaboration\Reference;
+
+use OCA\Talk\Chat\ChatManager;
+use OCA\Talk\Chat\MessageParser;
+use OCA\Talk\Exceptions\ParticipantNotFoundException;
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCP\Collaboration\Reference\IReference;
+use OCP\Collaboration\Reference\IReferenceProvider;
+use OCP\Collaboration\Reference\Reference;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+
+/**
+ * @psalm-type ReferenceMatch = array{token: string, message: ?int}
+ */
+class TalkReferenceProvider implements IReferenceProvider {
+	protected IURLGenerator $urlGenerator;
+	protected Manager $roomManager;
+	protected ChatManager $chatManager;
+	protected MessageParser $messageParser;
+	protected IL10N $l;
+	protected ?string $userId;
+
+	public function __construct(IURLGenerator $urlGenerator,
+								Manager $manager,
+								ChatManager $chatManager,
+								MessageParser $messageParser,
+								IL10N $l,
+								?string $userId) {
+		$this->urlGenerator = $urlGenerator;
+		$this->roomManager = $manager;
+		$this->chatManager = $chatManager;
+		$this->messageParser = $messageParser;
+		$this->l = $l;
+		$this->userId = $userId;
+	}
+
+
+	public function matchReference(string $referenceText): bool {
+		return $this->getTalkAppLinkToken($referenceText) !== null;
+	}
+
+	/**
+	 * @param string $referenceText
+	 * @return array|null
+	 * @psalm-return ReferenceMatch|null
+	 */
+	private function getTalkAppLinkToken(string $referenceText): ?array {
```

Comment:
I think that is good to put the rule to prevent private usage in the linter, not?!

This is a good method to cover with unit test because is very important to this class.

Reply:
Have tests prepared locally, will commit tomorrow